### PR TITLE
OBJLoader and OBJLoader2 allow empty/illegal smoothing group definition

### LINKED ISSUE #11707: OBJLoader + MTLLoader texture won`t work it
##### Description of the problem 

Hi,

I`m used to use three.js r53, so I want to upgrade to latest version(r86).

obj loader goes well, but when I try OBJLoader + MTLLoader that texture file does not work.

first, when I try as the sample(https://threejs.org/examples/webgl_loader_obj_mtl.html)

there have an error occurred [Unexpected line: 's']

I found out that because OBJLoader could not work with s(lineFirstChar), so I gave a exception for that.

second, after that I can view object, but textures do not work with it.

this is my site(http://3dbank.xyz:9021/viewer/viewer_hologram_u.do?conts_uid=2388)
##### Three.js version

- [ ] Dev
- [x] r86
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

### PR DESCRIPTION
**Pull request by kaisalmen**
Fixes #11707

### CODE REVIEW

**Comment by kaisalmen** [comment]
Hold on. I made a mistake and need to add another commit.

**Comment by kaisalmen** [comment]
Ok, now it is correct.

**Comment by WestLangley** [comment]
Again, what is your reason for expecting the loader to be fault-tolerant of illegal file formats? We can't do that. There are too many of them, no?

**Comment by mrdoob** [comment]
@WestLangley It is a problem when programs like ZBrush produce bad output. You're kind of forced to support it. Hopefully things will move to gltf in the future...

File: examples/js/loaders/OBJLoader.js [hunk 1]
```diff
@@ -589,7 +587,9 @@ THREE.OBJLoader = ( function () {
 
 					state.materialLibraries.push( line.substring( 7 ).trim() );
 
-				} else if ( ( result = smoothing_pattern.exec( line ) ) !== null ) {
+				} else if ( lineFirstChar === "s" ) {
+
+					result = line.length > 1 ? line.substring( 1 ).trim().split( " " ) : [];
```
Comment (mrdoob) [inline examples/js/loaders/OBJLoader.js]: How about just doing:

```js
result = line.split( ' ' );
```

And then

```js
if ( result.length > 1 ) {
	var value = result[ 1 ].trim().toLowerCase();
	state.object.smooth = ( value !== '0' && value !== 'off' );  
} else {
	state.object.smooth = true;
}
```

Comment (kaisalmen) [inline examples/js/loaders/OBJLoader.js]: Yes, thanks, that's better. The more complicated statement was a left over from detection by regex.

**Comment by mrdoob** [comment]
Thanks!

**Comment by kaisalmen** [comment]
@mrdoob, saw your comment after merge. The OBJ file was produced by:
`#Obj File : Exported From Geomagic (3D Systems Corporation)`
and not by ZBrush.
I am downloading a ZBrush Trial now. I guess, they export OBJ properly. I will find out.

**Comment by mrdoob** [comment]
Ah, that'll be good to know.

**Pull request merged**